app/src/main/java/com/example/plantsapp/PlantsFragment.kt

```diff
@@ -21,6 +26,20 @@ class PlantsFragment : Fragment() {
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.rvPlants.apply {
+            layoutManager = GridLayoutManager(context, 2)
+            adapter = PlantsAdapter().also { plantsAdapter ->
+
+                plantsViewModel.plants.observe(viewLifecycleOwner) {
+                    plantsAdapter.submitList(it)
+                }
+            }
+        }
+    }
```

Comment:
Suggest avoiding a lot of inner scopes: 
As I think - it is better to do 2 scopes - one for viewmodel , and one for binding 
- in view model you observe all things
- in view binding you create al lsetonclicklisteners and so on

Example (presenter is here is like viewmodel): 
![image](https://user-images.githubusercontent.com/23121846/137127637-985f06ca-d60d-479e-a5d7-b6b7b18c1604.png)